proposals/0088-enable-core-bpf-programs.md

```diff
@@ -0,0 +1,113 @@
+---
+simd: '0088'
+title: Enable Core BPF Programs
+authors:
+  - Joe Caulfield
+category: Standard
+type: Core
+status: Draft
+created: 2023-11-07
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+This proposal introduces the concept of Core BPF programs: programs which are
+essential to network operations. Currently, these exist as built-in programs
+known as "native" programs.
```

Comment:
I think this is a good shout. Technically, many programs are implemented as native/built-in for performance or privilege reasons. It just so happens these enhanced programs are often relied on by the cluster.

The naming convention for these programs should encapsulate:
1. The program may be relied on by the cluster, but may not be
2. The program may have certain privileges, but may not
3. The program may be non-upgradeable and changed via runtime feature gates, but it may not

In my opinion:
1. "essential" (or similar) captures true, not false
2. "enshrined" (or similar) captures true, not false
3. "special" (or similar) captures true, not false

I feel like "core" is broad enough to encapsulate all branches here. I'm happy to update the proposal to relax some of the wording around "essential to cluster operations" in favor of something similar to what I've written here.

What do you think?

Reply:
@ripatel-fd If we're good with "core" I think I'll resolve this one? Lmk, thanks.